pkg/agent/route/route_linux.go

```diff
@@ -115,19 +116,46 @@ func (c *Client) Initialize(nodeConfig *config.NodeConfig, done func()) error {
 func (c *Client) initIPTablesOnce(done func()) {
 	defer done()
 	backoffTime := 2 * time.Second
+	stopCh := signals.RegisterSignalHandlers()
+	c.syncIPTablesWithRetry(backoffTime)
+	klog.Info("Successfully synced iptables")
+	klog.Info("Starting sync loop for IPTables")
+	go c.ipTablesSyncLoop(stopCh)
+	return // FIXME: redundant
+}
+
+func (c *Client) ipTablesSyncLoop(stopCh <-chan struct{}) {
+	defer klog.Infof("Stopping IPTables SyncLoop")
+	var syncPeriod = 30 * time.Second
+	var syncTicker = time.NewTicker(syncPeriod)
+	defer syncTicker.Stop()
+	for {
```

Comment:
We typically use K8s utility functions like https://godoc.org/k8s.io/apimachinery/pkg/util/wait#Until for these scenarios

Reply:
Removed go's ticker with wait.Until. Updated the PR.